### PR TITLE
[com_fields] Add support for max length for text and textarea

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #14374.

### Summary of Changes
Adds the max length parameter to the text and textarea custom field. Additionally it fixes a bug in the maxlength parameter of the JLayout of the textarea field.

### Testing Instructions
- Create a text custom field for articles.
- Set the max length parameter to 2.
- Save an close the field.
- Edit an article.

### Expected result
It should not be possible to add more than 2 characters into the text field when editing the article.

### Actual result
The max length parameter is not available as it is a new feature.

### Documentation Changes Required
Adding the max length parameter description to the text and textarea field in the documentation pages.

### CODE REVIEW

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 4eccfc1f2ca2b0e133ecac7023112e5f104bdd13<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14458">issues.joomla.org/tracker/joomla-cms/14458</a>.</sub>

File: administrator/language/en-GB/en-GB.plg_fields_text.ini [hunk 1]
```diff
@@ -8,5 +8,7 @@ PLG_FIELDS_TEXT_LABEL="Text (%s)"
 PLG_FIELDS_TEXT_PARAMS_FILTER_LABEL="Filter"
 PLG_FIELDS_TEXT_PARAMS_FILTER_DESC="Allow the system to save certain html tags or raw data."
 PLG_FIELDS_TEXT_PARAMS_FILTER_RAW="Raw"
+PLG_FIELDS_TEXT_PARAMS_MAXLENGTH_LABEL="Max Length"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_fields_text.ini]: Maximum Length

This is more consistent with elsewhere

File: administrator/language/en-GB/en-GB.plg_fields_text.ini [hunk 2]
```diff
@@ -8,5 +8,7 @@ PLG_FIELDS_TEXT_LABEL="Text (%s)"
...
+PLG_FIELDS_TEXT_PARAMS_MAXLENGTH_DESC="Limits the number of characters that may be entered."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_fields_text.ini]: The maximum number of characters that can be entered.


This is more accurate and consistent with elsewhere

**Comment by laoneo** [comment]
Changed. Perhaps somebody should change the docs as well https://docs.joomla.org/Text_form_field_type.

**Comment by coolcat-creations** [comment]
I have tested this item :white_check_mark: successfully on d633af8bab15dde6b1aadde3b8f16cf20f5aae5d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14458">issues.joomla.org/tracker/joomla-cms/14458</a>.</sub>

**Comment by jeckodevelopment** [comment]
@laoneo can you please look at the conflicts?

**Comment by laoneo** [comment]
COnflicts solved.

**Comment by jeckodevelopment** [comment]
@coolcat-creations and @franz-wohlkoenig can you please re-test?

**Comment by laoneo** [comment]
Again conflicts solved, can we set this one to RTC?

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on b1e8a04f795266d8e4fd358418ac99fb3d0a6653<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14458">issues.joomla.org/tracker/joomla-cms/14458</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on b1e8a04f795266d8e4fd358418ac99fb3d0a6653<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14458">issues.joomla.org/tracker/joomla-cms/14458</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC as there are 2 successfully Tests?

**Comment by laoneo** [comment]
Just merged staging into this branch, can be set to RTC.

**Comment by normanmm** [comment]
I have tested this item :white_check_mark: successfully on aed12c46e79db080e906522d887a1852dbb8ab45<br /><br />After patch installation I can see a single query for all custom fields ( as opposed to a separate query for each custom field before the patch).
However ,I just want to point out ( though this is a different issue) that the fields are displayed quite messy and improperly , mostly in the front-end. This may  have to do with me not knowing properly the management of custom fields...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14458">issues.joomla.org/tracker/joomla-cms/14458</a>.</sub>

**Comment by normanmm** [comment]
Sorry , my mistake. My post was intended for #14558 .<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14458">issues.joomla.org/tracker/joomla-cms/14458</a>.</sub>

**Comment by laoneo** [comment]
Can we set this one to RTC as well?

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14458">issues.joomla.org/tracker/joomla-cms/14458</a>.</sub>

**Comment by infograf768** [comment]
@laoneo 

You have re-added obsolete language strings for both text an textarea plugins where we are now using Global strings
PR: https://github.com/joomla/joomla-cms/pull/15017

**Pull request merged**